### ISSUE TITLE
understanding the math calculations 

### ISSUE THREAD

**Issue by hassanelashiry** [Cyfrin/remix-fund-me-f23#11]
**Issue: Confusion About `MINIMUM_USD` Value in Solidity Code**

I noticed a potential discrepancy or confusion regarding the `MINIMUM_USD` value used in the contract. In the following line:

```solidity
uint256 public constant MINIMUM_USD = 5 * 10 ** 18;
```

The value is defined as `5 * 10 ** 18`, which represents 5 units of some currency, but it is crucial to note that this value is in Wei, not USD. The `fund()` function uses the `MINIMUM_USD` value in the `require` statement:

```solidity
function fund() public payable {
    require(msg.value.getConversionRate() >= MINIMUM_USD, "You need to spend more ETH!");
    addressToAmountFunded[msg.sender] += msg.value;
    funders.push(msg.sender);
}
```

Here, `getConversionRate()` converts the `msg.value` from Wei to USD with a factor of `1e18` because this function converts ETH to USD not Wei to USD. The `getConversionRate()` function is defined as follows:

```solidity
function getConversionRate(uint256 ethAmount) internal view returns (uint256) {
    uint256 ethPrice = getPrice();
    uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
    return ethAmountInUsd;
}
```

Since `msg.value` is in Wei and `getConversionRate()` converts it to USD (with a factor of `1e18` for precision), the result of `msg.value.getConversionRate()` is in USD, but multiplied by `1e18`.

Given this, if we are checking `msg.value.getConversionRate() >= MINIMUM_USD`, it appears that `MINIMUM_USD` should represent the minimum amount in USD (i.e., `5 * 1e18` USD) to match the precision used in the conversion function.

**To Summarize:**
- `MINIMUM_USD` is set as `5 * 10 ** 18`, which should be interpreted as `5 USD` in the context of the `getConversionRate()` function.
- `msg.value.getConversionRate()` returns USD multiplied by `1e18`, so `MINIMUM_USD` should indeed represent `5 USD` for this comparison to be accurate.

**This is just a discussion based on my understanding from the research I have done.

**Comment by cromewar** [comment]
Thanks for the detailed explanation! Your understanding is mostly correct, but there's a small clarification needed regarding the `MINIMUM_USD` value.

`MINIMUM_USD` is not in Wei but represents **5 USD** with 18 decimal precision. This precision is necessary to match the ETH-USD conversion logic, where conversion rates are scaled up by `1e18`. Everything else in your explanation is spot-on!

Appreciate your research and insight!

**Comment by hassanelashiry** [comment]
> Thanks for the detailed explanation! Your understanding is mostly correct, but there's a small clarification needed regarding the `MINIMUM_USD` value.
> 
> `MINIMUM_USD` is not in Wei but represents **5 USD** with 18 decimal precision. This precision is necessary to match the ETH-USD conversion logic, where conversion rates are scaled up by `1e18`. Everything else in your explanation is spot-on!
> 
> Appreciate your research and insight!


Thank you for the clarification!
